packages/sw-background-sync-queue/src/lib/queue.js

```diff
@@ -0,0 +1,143 @@
+import stringHash from 'string-hash';
+import idbQHelper from './idbHelper';
+import requestManager from './requestManager';
+import bcmanager from './broadcastManager';
+
+let _queue;
+let _config;
+
+/**
+ * Core queue class that handles all the enqueue and dequeue
+ * as well as cleanup code for the background sync queue
+ * @class Queue
+ */
+class Queue {
+	/**
+	 * Creates an instance of Queue.
+	 *
+	 * @memberOf Queue
+	 */
+	constructor() {
+		this.initialize();
+	}
+
+	/**
+	 * initialize the queue object from the idb
+	 * giving access to any penging queues
+	 * @param {Object} config config that gives values like
+	 * maxAge of requests in iDB
+	 *
+	 * @memberOf Queue
+	 */
+	async initialize(config) {
+		_config = config;
+		_queue = await idbQHelper.get('queue') || [];
+	}
+
+	/**
+	 * push any request to background sync queue which would be played later
+	 * prefferably when network comes back
+	 *
+	 * @param {Request} request request object to be queued by this
+	 * @param {Object} config optional config to override config params
+	 *
+	 * @memberOf Queue
+	 */
+	async push(request, config) {
+		let localConfig = Object.assign({}, _config, config);
```

Comment:
actually I think it'll be a good idea to give override to some particular requests.
may be these requests are more critical and i want them to live for a longer time even if they fail. Or may be I want their response to live longer.
Comments?